core/synch.c

```diff
@@ -1749,6 +1749,14 @@ translate_from_synchall_to_dispatch(thread_record_t *tr, thread_synch_state_t sy
                 arch_mcontext_reset_stolen_reg(dcontext, mc);
             }
         });
+        IF_AARCHXX({
+            // XXX i#4495: Consider saving stolen reg's application value.
+            set_stolen_reg_val(mc, (reg_t)os_get_dr_tls_base(dcontext));
```

Comment:
I think this code will make more sense if it's down after we point pc at get_reset_exit_stub.  It's directly related to that step.  If we changed our strategy to do sthg different we might not want this at new code at all: so best to keep it coupled to that.  I would also include a comment stating *why* we have to do this.

Reply:
Done.